user/languages/java.md

```diff
@@ -4,94 +4,120 @@ layout: en
 
 ---
 
+<div id="toc">
+</div>
+
+<aside markdown="block" class="ataglance">
+
+| Java                         | Default                                                                                                              |
+|:-----------------------------|:---------------------------------------------------------------------------------------------------------------------|
+| Default `install`            | [Gradle](#Gradle-Dependency-Management), [Maven](#Maven-Dependency-Management), [Ant](#Ant-Dependency-Management)    |
+| Default `script`             | [Gradle](#Gradle-Default-Script-Command), [Maven](#Maven-Default-Script-Command), [Ant](#Ant-Default-Script-Command) |
+| [Matrix keys](#Build-Matrix) | `jdk`, `env`                                                                                                         |
+| Support                      | [Travis CI](mailto:support@travis-ci.com)                                                                            |
+
+Minimal example:
+
+```yaml
+  language: java
+```
+</aside>
+
 ### What This Guide Covers
 
-This guide covers build environment and configuration topics specific to Java projects. Please make sure to read our [Getting Started](/user/getting-started/) and [general build configuration](/user/customizing-the-build/) guides first.
+{{ site.data.snippets.trusty_note }}
 
-<div id="toc"></div>
+The rest of this guide covers configuring Java projects in Travis CI. If you're
+new to Travis CI please read our [Getting Started](/user/getting-started/) and
+[build configuration](/user/customizing-the-build/) guides first.
 
 ## Overview
 
-The Travis CI environment provides recent versions of Oracle JDK 8 (default), Oracle JDK 9, OpenJDK 7, Gradle 2.0, Maven 3.2 and Ant 1.8, and has sensible defaults for projects that use Gradle, Maven or Ant.
+The Travis CI environment contains various versions of OpenJDK, Oracle JDK,
+Gradle, Maven and Ant.
 
-To use the Java environment add the following to your `.travis.yml`:
+To use the Java environment, add the following to your `.travis.yml`:
 
 ```yaml
 language: java
 ```
 {: data-file=".travis.yml"}
 
-Older OpenJDK and Oracle JDK versions may be available on [Precise](/user/reference/precise#JDK).
-
 ## Projects Using Maven
 
-### Default script Command
```

Comment:
This and the following sections look all messed up in the diff because I moved the "Dependency Management" before "Script command", since `install` runs before `script`.